AzureRmStorageTableCoreHelper.psm1

```diff
@@ -42,23 +42,30 @@ function ExecuteQueryAsync
 		[Parameter(Mandatory=$true)]
 		$Table,
 		[Parameter(Mandatory=$true)]
-		$TableQuery
+		$TableQuery,
+		[Parameter()]
+		$Paginate = $true
 	)
 	# Internal function 
 	# Executes query in async mode
 
 	if ($TableQuery -ne $null)
 	{
         $token = $null
-		$AllRows = @()
-		do
+		if ($Paginate)

```

Comment:
The general guideline is to always use continuation token while invoking Segmented Query, and it's recommended to always use Segmented Query.
The reason is when server (both storage and cosmosdb) clusters data into multiple physical partitions, the first page may contain no results but a continuation token.
I don't think we need this change, and I don't see any benefit of doing this change either. User may get confused what value to give to the Paginate parameter and ends up using true anyways.